### ISSUE TITLE
[Bug] TXTP/TXTH file issues (including other subformats ie. adpcm) when adding to Foobar2000 from archive

### ISSUE THREAD

**Issue by Level99OCR** [vgmstream/vgmstream#1044]
Foobar2000 version: 1.6.8
vgmstream plugin version: r1690-35-gc38c09fb (Dec 25 2021)

Notes: All of my chiptunes are in either .zip or .7z format for space conservation. 
Example files: 
Air (2002-08-08)(Key)(-)(NEC Interchannel)[PS2] (.adpcm, .vgmstream, .txth, .TBL)
Pokemon Brilliant Diamond + Shining Pearl [Pocket Monsters Brilliant Diamond + Shining Pearl] (v1.1.0)(2021-11-19)(ILCA)(Nintendo, The Pokemon Company)[Switch] (.wem, .bnk, .txtp)

Expected behavior: Dragging and dropping the archives from Windows Explorer into a playlist should scan then add the appropriate supported files with correct track length, and be playable.
Actual behavior: Dragging and dropping the archives from Windows Explorer into a playlist does scan, albeit much more slowly than if uncompressed. When completed, items with txtp, txth, wem, bnk, adpcm, and vgmstream file extensions are added but only some have song length. Those without songlength are unplayable.

Playable: wem, bnk (items from bnk scans listed individually as they should be)
Unplayable: txtp, adpbm, txth, vgmstream

If the files are added from those exact archives but they've been uncompressed into folders, the scanning takes a fraction of the time as the compressed archive and almost all files are added correctly and are playable (the exception being txth, but as far as I'm aware those aren't supposed to contain any playable media themselves).

**Comment by kode54** [comment]
This shouldn't be expected to work flawlessly anyway. A lot of the (definitely not chiptunes) archives on a certain site are still solid compressed, which means that for each .txth or .txtp indexed, it has to separately unpack the entire archive a second time for the referenced files.

The archive readers were never meant to handle files this large.

**Comment by Level99OCR** [comment]
Any archives I add to Foobar2000 are either zip or 7z non-solidblock exactly to avoid the issue you mentioned (also because the scanning time is exponentially worse). So in any of my examples please assume that the archives are non-solidblock.

Also I'm not entirely sure what size files the archive readers were meant to deal with in the first place, but the Air (2002-08-08)(Key)(-)(NEC Interchannel)[PS2].zip is 171MB in size. In contrast, I just dragged the zip for Aa! Megami-sama! (2007-02-22)(-)(Marvelous)[PS2] onto Foobar2000 which is 412MB in size and all contents scan and play properly. Unsure how this factors into archive file size limitations in this regard unless I'm missing something.

Honestly I'd like to focus on just the Air PS2 files to start, maybe it'll help narrow down what could be happening in my case.

Here are the contents of the archive:
![image](https://user-images.githubusercontent.com/6171738/147826635-3f9ba779-abbb-419f-87a1-d2dce4428b26.png)

Dragging and dropping the zip file onto Foobar2000 gives the following entries after scanning:
![image](https://user-images.githubusercontent.com/6171738/147826679-f0760e36-1963-4d12-b42f-ffeb2703329c.png)
None of these files are playable as a result.

Dragging and dropping the extracted folder onto Foobar2000 gives the following entries after scanning:
![image](https://user-images.githubusercontent.com/6171738/147826735-7fbd906d-5c78-4182-81fe-0d00a7b156f5.png)
The .adpcm and .vgmstream files of course are still invalid, but the Duration data, Sample Frequency and Sample Bits all show up for the remaining tracks and are also all properly playable.

**Comment by bnnm** [comment]
Should work now (albeit slowly, since each .adpcm needs to read the .txth and .names sub-files, and the archive component isn't probably well tuned for that).

**Comment by Level99OCR** [comment]
Fantastic, thank you. I will wait for the next automated build to compile and test as soon as I can, and let you know the results.

**Comment by Level99OCR** [comment]
Success on at least one of the issues I experienced! The ADPCM files in Air now add from archive and play without issue.
![image](https://user-images.githubusercontent.com/6171738/147834154-aa951f0f-dfa3-4e96-8e82-32061bf9f287.png)
So TXTH seems to work. And I tested TXTP from Xevious 3DG+ and they also seemed to have added without issue (though for some reason it added double tracks: one set from TXTP and another from scanning the raw XA file. Unsure if there's any way to flag the scanner to not scan the XA if TXTP files are present?)
![image](https://user-images.githubusercontent.com/6171738/147834249-b1e694a7-1bc1-4e88-b77e-da2841727a18.png)
The TXTP files for Pokemon Brilliant Diamond + Shining Pearl still seem to have issues being scanned from within archives:
![image](https://user-images.githubusercontent.com/6171738/147834276-04f6e8ad-3187-42d3-88c2-1c5fdc3f3b7b.png)
Unsure if the TXTP issue with Pokemon is due to the rip or vgmstream.
